File: update_layout.py
import re

filepath = 'app/layout.tsx'

with open(filepath, 'r') as f:
    content = f.read()

# Make sure we import ThemeProvider
if 'ThemeProvider' not in content:
    content = content.replace('import { RouteHandler } from "@/components/route-handler";',
        'import { RouteHandler } from "@/components/route-handler";\nimport { ThemeProvider } from "@/components/theme-provider";')

# Wrap RouteHandler and children with ThemeProvider
body_pattern = r'(<body className="[^"]*">)\s*<Suspense fallback=\{null\}>\s*<RouteHandler />\s*</Suspense>\s*\{children\}\s*</body>'

replacement = r'''\1
        <ThemeProvider attribute="class" defaultTheme="dark" enableSystem disableTransitionOnChange>
          <Suspense fallback={null}>
            <RouteHandler />
          </Suspense>
          {children}
        </ThemeProvider>
      </body>'''

content = re.sub(body_pattern, replacement, content)

with open(filepath, 'w') as f:
    f.write(content)
print("Updated app/layout.tsx")

File: update_layout.js
const fs = require('fs');

const path = 'app/layout.tsx';
let layout = fs.readFileSync(path, 'utf8');

// Add imports
layout = layout.replace(
  'import { ThemeProvider } from "@/components/theme-provider";',
  'import { ThemeProvider } from "@/components/theme-provider";\nimport { ChatProvider } from "@/components/chat-context";\nimport { ChatInterface } from "@/components/chat-interface";'
);

// Wrap children
layout = layout.replace(
  '<ThemeProvider attribute="class" defaultTheme="dark" enableSystem disableTransitionOnChange>',
  '<ThemeProvider attribute="class" defaultTheme="dark" enableSystem disableTransitionOnChange>\n          <ChatProvider>'
);

layout = layout.replace(
  '{children}\n        </ThemeProvider>',
  '{children}\n            <ChatInterface />\n          </ChatProvider>\n        </ThemeProvider>'
);

fs.writeFileSync(path, layout);
